62.CAD/fragment_shader.hlsl

```diff
@@ -15,6 +16,68 @@ void endInvocationInterlockEXT();
 // Write a general one, and maybe another one that uses precomputed values, and move these to somewhere nice in our builtin hlsl shaders if we don't have one
 // See: https://github.com/erich666/GraphicsGems/blob/master/gems/Roots3And4.c
 
+template<typename float_t, typename CurveType>
+struct LineStyleClipper
+{
+    using float2_t = vector<float_t, 2>;
+
+    static LineStyleClipper<float_t, CurveType> construct(uint32_t styleIdx, 
+                                                          CurveType curve,
+                                                          typename CurveType::ArcLenCalculator arcLenCalc,
+                                                          float totalArcLen)
+    {
+        LineStyleClipper<float_t, CurveType> ret = { styleIdx, curve, arcLenCalc, totalArcLen };
+        return ret;
+    }
+    
+    struct ArrayAccessor
+    {
+        uint32_t styleIdx;
+        using value_type = float;
+
+        float operator[](const uint32_t ix)
+        {
+            return lineStyles[styleIdx].stipplePattern[ix];
+        }
+    };
+    
+    float2_t operator()(const float t)
+    {
+        const float arcLen = arcLenCalc.calcArcLen(t);
+        float_t tMappedToPattern = frac(arcLen / float(globals.screenToWorldRatio) * lineStyles[styleIdx].recpiprocalStipplePatternLen + lineStyles[styleIdx].phaseShift);

```

Comment:
Take `arcLen / float(globals.screenToWorldRatio)` out, name it const float worldSpaceArcLen

Reply:
also now that you have both screen to world and world to screen ratio in globals; why do you prefer division over multiplication?

Reply:
Also rename "tMappedToPattern" to "normalizedPlaceInPattern"

Reply:
IMHO all this stuff to do with finding the patternIndex (from `tMappedToPattern` to `patternIdx`) better be wrapped up in a `StyleIndexFinder<StyleAccessor>` with a `operator(const StyleAccessor::float_t patternSpaceArcLength)`, you might need it later down the line to do other things

You'd feed `worldSpaceArcLen` to this `operator()`